addons/stock/models/stock_quant.py

```diff
@@ -341,6 +429,46 @@ def _quant_tasks(self):
         self._merge_quants()
         self._unlink_zero_quants()
 
+    def write(self, vals):
+        if 'inventory_quantity' in vals and not self.env.context.get('dont_create_inventory_line'):
```

Comment:
i'm not sure it shouldn't be the other way around, here we don't patch write if `dont_create_inventory_line` is not in the context

i think it would be preferable to do all the weird stuff only if a context key is present and added by the view

what do you think?